alchemy/src/cloudflare/snippet.ts

```diff
@@ -0,0 +1,466 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { extractCloudflareResult } from "./api-response.ts";
+import {
+  createCloudflareApi,
+  type CloudflareApi,
+  type CloudflareApiOptions,
+} from "./api.ts";
+import { findZoneForHostname, type Zone } from "./zone.ts";
+
+/**
+ * Properties for creating or updating a Snippet
+ */
+export interface SnippetProps extends CloudflareApiOptions {
+  /**
+   * The zone to create the snippet in
+   */
+  zone: string | Zone;
+
+  /**
+   * The identifying name of the snippet
+   * Must be unique within the zone
+   * Only lowercase letters (a-z), numbers (0-9), and underscores (_) are allowed
+   */
+  name: string;
+
+  /**
+   * The JavaScript code content of the snippet
+   */
+  content: string;
+
+  /**
+   * Whether to delete the snippet
+   * If set to false, the snippet will remain but the resource will be removed from state
+   * @default true
+   */
+  delete?: boolean;
+
+  /**
+   * Whether to adopt an existing snippet
+   * @default false
+   */
+  adopt?: boolean;
+
+  /**
+   * The Cloudflare-generated zone ID (only used for update/delete operations)
+   * @internal
+   */
+  zoneId?: string;
+
+  /**
+   * The snippet name (only used for update/delete operations)
+   * @internal
+   */
+  snippetId?: string;
```

Comment:
This should not be in the props, right? We have it in state and it should be encapsulated.